src/transformers/modeling_utils.py

```diff
@@ -1624,11 +1625,30 @@ def can_generate(cls) -> bool:
         Returns:
             `bool`: Whether this model can generate sequences with `.generate()`.
         """
-        # Detects whether `prepare_inputs_for_generation` has been overwritten, which is a requirement for generation.
-        # Alternativelly, the model can also have a custom `generate` function.
-        if "GenerationMixin" in str(cls.prepare_inputs_for_generation) and "GenerationMixin" in str(cls.generate):
-            return False
-        return True
+        # Directly inherits `GenerationMixin` -> can generate
+        if "GenerationMixin" in str(cls.__bases__):
+            return True
+        # Model class overwrites `generate` -> can generate
+        if str(cls.__class__.__name__) in str(cls.generate):
```

Comment:
if it does not inherit from generationMixin should we rais an error?

Reply:
uhmm, you're right, the previous check confirmed that both `prepare_inputs_for_generation` AND `generate` were overwritten

(I was thinking about the time series models, which can call their custom `generate`)

Reply:
Not quite, we have the time series models that overwrite `generate` (and, prior to this changes, `can_generate()` returned `True` for them)